torchtune/models/flamingo/_component_builders.py

```diff
@@ -197,21 +527,34 @@ def flamingo_decoder(
     for idx in range(1, num_layers + 1):
 
         # Self attention layers for text decoder
-        rope = RotaryPositionalEmbeddings(dim=head_dim, max_seq_len=max_seq_len, base=rope_base)
-        self_attn = MultiHeadAttention(
+        rope = Llama3ScaledRoPE(dim=head_dim, max_seq_len=max_seq_len, base=rope_base)
```

Comment:
rope should be instantiated only once, ouside of the for loop, to avoid copies and extra memory